include/irr/asset/IMeshPacker.h

```diff
@@ -21,13 +18,96 @@ template <typename MeshBufferType>
+        //m_vtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[0], 0u, 0u, 0u, allocParams.vertexBuffSupportedSizeInBytes / allocParams.vertexbufferMinAllocSize, allocParams.vertexbufferMinAllocSize);
+
+        //m_alctrReservedSpace[1] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_idxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[1], 0u, 0u, 0u, allocParams.indexBuffSupportedSizeInBytes / allocParams.indexBufferMinAllocSize, allocParams.indexBufferMinAllocSize);
+
+        //m_alctrReservedSpace[2] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_perInsVtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[2], 0u, 0u, 0u, allocParams.perInstanceVertexBuffSupportedSizeInBytes / allocParams.perInstanceVertexbufferMinAllocSize, allocParams.perInstanceVertexbufferMinAllocSize);
+        
+
+
+        //init MDI data allocator
+        m_alctrReservedSpace[3] = static_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<size_t>::reserved_size(alignof(uint32_t), allocParams.MDIDataSizeInBytes / sizeof(DrawElementsIndirectCommand_t), 1u), _IRR_SIMD_ALIGNMENT));
+        m_MDIDataAlctr = core::GeneralpurposeAddressAllocator<size_t>(m_alctrReservedSpace[3], 0u, 0u, alignof(uint32_t), allocParams.MDIDataSizeInBytes / sizeof(DrawElementsIndirectCommand_t), /*i git*/1u);
+
+        //1 attrib enabled == 1 binding
+        for (uint16_t attrBit = 0x0001, location = 0; location < 16; attrBit <<= 1, location++)
+        {
+            if (m_outVtxInputParams.enabledAttribFlags & attrBit)
+            {
+                m_outVtxInputParams.bindings[location].stride = getTexelOrBlockBytesize(static_cast<E_FORMAT>(m_outVtxInputParams.attributes[location].format));
+                m_outVtxInputParams.bindings[location].inputRate = preDefinedLayout.bindings[preDefinedLayout.attributes[location].binding].inputRate;
+            }
+        }
+    }
+
+    virtual std::optional<PackedMeshBufferData> alloc(const MeshBufferType* meshBuffer) = 0;
+    virtual PackedMeshBuffer<MeshBufferType> commit(const core::vector<std::pair<const ICPUMeshBuffer*, PackedMeshBufferData>>& meshBuffers) = 0;
+
+    asset::ICPUBuffer* getMultiDrawIndirectBuffer() { return m_outMDIData.get(); }
+
+protected:
+    virtual ~IMeshPacker() 
+    {
+        for (int i = 0; i < 4; i++)
+            _IRR_ALIGNED_FREE(m_alctrReservedSpace[i]);
+    }
 
 protected:
-    virtual ~IMeshPacker() {}
+    //output mesh buffers data
+    SVertexInputParams m_outVtxInputParams;
+
+    uint8_t* m_alctrReservedSpace[4];
+    core::GeneralpurposeAddressAllocator<size_t> m_vtxBuffAlctr;
+    core::GeneralpurposeAddressAllocator<size_t> m_idxBuffAlctr;
+    core::GeneralpurposeAddressAllocator<size_t> m_perInsVtxBuffAlctr;
+    core::GeneralpurposeAddressAllocator<size_t> m_MDIDataAlctr;
```

Comment:
use `uint32_t` address spaces, the allocators will use less reserved space if they don't have to do arithmetic in 64bit